src/components/SuggestToWeb.js

```diff
@@ -0,0 +1,221 @@
+      if (match === 7) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll7Matches.push(combineCharacterInfo[index]);
+      } else if (match === 6) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll6Matches.push(combineCharacterInfo[index]);
+      } else if (match === 5) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll5Matches.push(combineCharacterInfo[index]);
+      } else if (match === 4) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll4Matches.push(combineCharacterInfo[index]);
+      } else if (match === 3) {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match}`);
+        arrayOfAll3Matches.push(combineCharacterInfo[index]);
+      } else {
+        // //console.log(`This array: ${combineCharacterInfo[index]} matches this many times: ${match} :(`)
+      }
+    });
+    setCharactersWith7Matches(arrayOfAll7Matches);
+    setCharactersWith6Matches(arrayOfAll6Matches);
+    setCharactersWith5Matches(arrayOfAll5Matches);
+    setCharactersWith4Matches(arrayOfAll4Matches);
+    setCharactersWith3Matches(arrayOfAll3Matches);
+  }
+  return (
+    <div className="h-full my-2">
+
+      <Web webOfTeam = {props.webOfTeam}/>
+
+      <div className="h-[60vh]row-span-2 p-2">
+        <div className="flex flex-col h-full justify-end">
+          {/* <h3>Characters with 7 Links:</h3>
+          <div className="grid grid-cols-5 max-w-full h-[120px] bg-gradient-radial from-purple-200 via-purple-100 to-purple-50 m-1 p-2 shadow-[inset_0_-5px_6px_rgba(0,0,0,0.6)] border-2 border-slate-900 overflow-auto">
+            {charactersWith7Matches &&
+              charactersWith7Matches.map((character) => (
+                <div key={character} onClick= {() => {
+                  newArraySuggestion(character);
+                  props.handleNewDetails(character)
+                  console.log(character);
+                }}>
+                  <SingleCard characterId={character[0]} characterThumb = {character[1]} characterArt = {character[2]} characterType={character[3]} characterRarity={character[4]}/>
+                </div>
+              ))}
+          </div> */}
+          <h3>Characters with 6 Links:</h3>
+          <div className="grid grid-cols-5 h-[120px] max-w-full bg-gradient-radial from-purple-200 via-purple-100 to-purple-50  m-1 p-2 shadow-[inset_0_-5px_6px_rgba(0,0,0,0.6)] border-2 border-slate-900 overflow-auto">
+            {charactersWith6Matches &&
+              charactersWith6Matches.map((character) => (
+                <div key={character} onClick= {() => {
+                  newArraySuggestion(character)
+                  props.handleNewDetails(character);
```

Comment:
This is great - perfect use of passing a handler to a subcomponent. However, if you look at the flow of data, this means that when you click a character here, you'll update the characterDetails state in AllComponents which will then change the props getting passed down to this component. So that means you don't need the selectedCharacterLinks as state either, it's just derived from the prop!

To rephrase, the `newArraySuggestion` function is unnecessary and should be deleted